### PR TITLE
[CBRD-22976][replication] system op with replication confirmation

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/CBRD-22976

Add log_sysop_commit_replicated function; it should be preceded by a HA stream append and log its stream position.

During recovery undo, the sysop is considered committed only if stream position is confirmed to be replicated.

Rollback is handled like a commit, it remains to be determined.

### CODE REVIEW

File: src/transaction/log_record.hpp [hunk 1]
```diff
@@ -320,6 +322,9 @@ struct log_rec_sysop_end
       bool is_sysop_postpone;	/* true if run postpone is used during a system op postpone, false if used during
 				 * transaction postpone */
     } run_postpone;		/* run postpone info */
+    // *INDENT-OFF*
```
Comment (razvanradulescu) [inline src/transaction/log_record.hpp]: what about moving it before the union ?
(easier debugging)

Comment (ribram) [reply src/transaction/log_record.hpp]: It is part of the union; each type of sysop end has different data.

File: src/transaction/log_global.c [hunk 2]
```diff
@@ -73,6 +73,7 @@ log_global::log_global ()
      }
   , group_commit_info LOG_GROUP_COMMIT_INFO_INITIALIZER
   , m_tran_complete_mgr (NULL)    // to be instantiated on boot
+  , m_ack_stream_position (std::numeric_limits<cubstream::stream_position>::max ())
```
Comment (alexstanro) [inline src/transaction/log_global.c]: m_ack_stream_position should be 0

File: src/transaction/log_recovery.c [hunk 3]
```diff
@@ -4849,6 +4850,19 @@ log_recovery_undo (THREAD_ENTRY * thread_p)
 		      /* compensate */
 		      LSA_COPY (&prev_tranlsa, &sysop_end->compensate_lsa);
 		    }
+		  else if (sysop_end->type == LOG_SYSOP_END_COMMIT_REPLICATED)
```
Comment (alexstanro) [inline src/transaction/log_recovery.c]: I expect similar code in log_rollback

File: src/transaction/log_global.c (hunk 2 above)
Comment (ribram) [reply src/transaction/log_global.c]: it is temporary max to make recovery work with commit replicated sysops. should be properly initialized when m_ack_stream_position is management is mature.

File: src/transaction/log_recovery.c (hunk 3 above)
Comment (ribram) [reply src/transaction/log_recovery.c]: fixed

**Pull request merged**